src/Paket.Core/Dependencies/Nuget.fs

```diff
@@ -1,134 +1,548 @@
+                if verbose then
+                    verbosefn "%O %O already downloaded." packageName version
+            elif not force && getFromCache caches then
+                ()
+            else
+                match source with
+                | LocalNuGet(path,_) ->
+                    let path = Utils.normalizeLocalPath path
+                    let di = Utils.getDirectoryInfoForLocalNuGetFeed path alternativeProjectRoot root
+                    let nupkg = NuGetLocal.findLocalPackage di.FullName packageName version
+                    
+                    use _ = Profile.startCategory Profile.Category.FileIO
+                    File.Copy(nupkg.FullName,targetFileName)
+                | _ ->
+                // discover the link on the fly
+                let downloadUrl = ref ""
                 try
-                    let cachedObject = JsonConvert.DeserializeObject<NuGetPackageCache> json
+                    if authenticated then
+                        tracefn "Downloading %O %O%s" packageName version (if groupName = Constants.MainDependencyGroup then "" else sprintf " (%O)" groupName)
+                    let! nugetPackage = GetPackageDetails alternativeProjectRoot root force [source] groupName packageName version
+
+                    let encodeURL (url:string) = url.Replace("+","%2B")
+                    let downloadUri =
+                        if Uri.IsWellFormedUriString(nugetPackage.DownloadLink, UriKind.Absolute) then
+                            Uri nugetPackage.DownloadLink
+                        else
+                            let sourceUrl =
+                                if nugetPackage.Source.Url.EndsWith("/") then nugetPackage.Source.Url
+                                else nugetPackage.Source.Url + "/"
+                            Uri(Uri sourceUrl, nugetPackage.DownloadLink)
+
+                    downloadUrl := downloadUri.ToString()
+
+                    if authenticated && verbose then
+                        tracefn "  from %O" !downloadUrl
+                        tracefn "  to %s" targetFileName
                     
-                    if (PackageName cachedObject.PackageName <> packageName) ||
-                      (cachedObject.Version <> version.Normalize())
-                    then
-                        traceVerbose (sprintf "Invalidating Cache '%s:%s' <> '%s:%s'" cachedObject.PackageName cachedObject.Version packageName.Name (version.Normalize()))
-                        cacheFile.Delete()
-                        None
+                    use trackDownload = Profile.startCategory Profile.Category.NuGetDownload
+                    let! license = Async.StartChild(DownloadLicense(root,force,packageName,version,nugetPackage.LicenseUrl,licenseFileName), 5000)
+
+                    let request = HttpWebRequest.Create(downloadUri) :?> HttpWebRequest
+#if NETSTANDARD1_6
+                    // Note: this code is not working on regular non-dotnetcore
```

Comment:
Yes. It's really unfortunate. I have the same code in FAKE as well :/

Problem is this is already a lightweight abstraction (see Paket.Utils) it's just that at this particular place they have subtle different behavior. I noted somewhere I'd would be better to use the new `HttpClient` everywhere but I didn't want to change everything at the time....